watcher/src/utils/near.ts

```
@@ -22,6 +24,32 @@ export const getNearProvider = async (rpc: string): Promise<Provider> => {
   return provider;
 };
 
+export async function getTimestampByBlock(
+  provider: Provider,
+  blockHeight: number
+): Promise<number> {
+  const block: BlockResult = await fetchBlockByBlockId(provider, blockHeight);
+  return block.header.timestamp;
+}
+
+export async function fetchBlockByBlockId(
+  provider: Provider,
+  blockHeight: BlockId
+): Promise<BlockResult> {
+  let success: boolean = false;
+  let block: BlockResult = {} as BlockResult;
+  while (!success) {
+    try {
+      block = await provider.block({ blockId: blockHeight });
+      success = true;
+    } catch (e) {
+      console.error('Error fetching block', e);
+      await sleep(5000);
+    }
+  }
```

should we have a condition for max retries or max time?

---

I thought about that.  But there is really nothing else we can do but wait.  The error message every 5 seconds will show up in my logging monitoring.  So, it becomes more apparent.  In other places, we have exponential backoff.  But what happens is that it backs off so much that the errors get lost.